core/src/main/java/org/kiwix/kiwixmobile/core/help/HelpFragment.kt

```diff
@@ -46,38 +51,45 @@ class HelpActivity : BaseActivity() {
     }
   }
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_help)
+  override fun inject(baseActivity: BaseActivity) {
+    (baseActivity as CoreMainActivity).cachedComponent.inject(this)
+  }
 
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
+    super.onActivityCreated(savedInstanceState)
+    val activity = requireActivity() as AppCompatActivity
     activity_help_feedback_text_view.setOnClickListener { sendFeedback() }
     activity_help_feedback_image_view.setOnClickListener { sendFeedback() }
     diagnostic_clickable_area.setOnClickListener { sendDiagnosticReport() }
-    setSupportActionBar(toolbar)
-    toolbar.setNavigationOnClickListener { onBackPressed() }
-    supportActionBar?.let {
+    activity.setSupportActionBar(toolbar)
```

Comment:
don't we have to set this to null in onDestroy? Or what was that memory leak fix?

Reply:
Yes, it caused some errors. I should go back and look at that. Some odd things are happening in the historybookmarks-to-fragment branch because of the toolbar being null..